src/sagas/networkSettings.js

```diff
@@ -0,0 +1,118 @@
+import { NETWORK_SETTINGS_STATUS } from '../constants';
+import { types, isVersionAllowedUpdate, selectToken, setNetworkSettingsStatus } from '../actions';
+import { all, call, put, select, take, takeEvery } from 'redux-saga/effects';
+import hathorLib from '@hathor/wallet-lib';
+import { getGlobalWallet } from '../modules/wallet';
+import helpers from '../utils/helpers';
+import walletUtils from '../utils/wallet'
+
+/**
+ * Change network settings with new data
+ * @param {Object} action with request payload.
+ * @param {string} action.data.node
+ * @param {string} action.data.txMining
+ * @param {string} action.data.explorer
+ * @param {string} action.data.explorerService
+ * @param {string} action.data.walletService
+ * @param {string} action.data.walletServiceWS
+ * @param {string} action.pin
+ */
+export function* changeNetworkSettings({ data, pin }) {
+  yield put(setNetworkSettingsStatus({ status: NETWORK_SETTINGS_STATUS.LOADING }));
+  const useWalletService = yield select((state) => state.useWalletService);
+  const wallet = getGlobalWallet();
+  const currentServer = useWalletService ?
+    hathorLib.config.getWalletServiceBaseUrl() :
+    hathorLib.config.getServerUrl();
+
+  const currentWsServer = useWalletService ?
+    hathorLib.config.getWalletServiceBaseWsUrl() :
+    '';
+
+  const currentNetwork = wallet.getNetwork();
+
+  // Update new server in storage and in the config singleton
+  wallet.changeServer(data.node);
+
+  // We only have a different websocket server on the wallet-service facade, so update the config singleton
+  if (useWalletService) {
+    yield call([wallet, wallet.changeWsServer], newWsServer);
+  }
+
+  let versionData;
+  try {
+    versionData = yield call([wallet, wallet.getVersionData]);
+  } catch (e) {
+    // Invalid node, so go back to the previous server and return
+    wallet.changeServer(currentServer);
+    if (useWalletService) {
+      yield call([wallet, wallet.changeWsServer], currentWsServer);
+    }
+    yield put(setNetworkSettingsStatus({ status: NETWORK_SETTINGS_STATUS.ERROR, error: t`Invalid node.` }));
+    return;
+  }
+
+  if (versionData.network === 'mainnet') {
+    yield executeNetworkSettingsUpdate({ ...data, network: versionData.network }, pin);
+    return;
+  }
+
+  // We must get user confirmation because the new network is a testnet or privatenet
+  const newNetwork = hathorLib.helpersUtils.getNetworkFromFullNodeNetwork(versionData.network);
+  yield put(setNetworkSettingsStatus({ status: NETWORK_SETTINGS_STATUS.WAITING_NETWORK_CONFIRMATION, newNetwork }));
+
+  // Wait until network settings status changes
+  yield take(types.NETWORKSETTINGS_SET_STATUS);
```

Comment:
This will wait until the status change to `NETWORK_CONFIRMED` or `READY` meaning that he navigated out of the screen, right?

Maybe add a comment on this here

Reply:
Done https://github.com/HathorNetwork/hathor-wallet/pull/701/commits/d031d540bc1d02e5064ffd4b781db78c96ce4fa3